packages/job-worker/src/blueprints/postProcess.ts

```diff
@@ -315,21 +316,25 @@ export function postProcessBucketAdLib(
 	blueprintId: BlueprintId,
 	bucketId: BucketId,
 	rank: number | undefined,
-	importVersions: RundownImportVersions
+	importVersions: RundownImportVersions,
+	uniquenessId: string | undefined
 ): BucketAdLib {
+	const id: PieceId = protectString(
+		getHash(
+			`${innerContext.showStyleCompound.showStyleVariantId}_${innerContext.studioIdProtected}_${bucketId}_bucket_adlib_${externalId}`
+		)
+	)
 	const piece: BucketAdLib = {
 		...itemOrig,
 		content: omit(itemOrig.content, 'timelineObjects'),
-		_id: protectString(
-			getHash(
-				`${innerContext.showStyleCompound.showStyleVariantId}_${innerContext.studioIdProtected}_${bucketId}_bucket_adlib_${externalId}`
-			)
-		),
+		_id: id,
 		externalId,
 		studioId: innerContext.studioIdProtected,
+		showStyleBaseId: innerContext.showStyleCompound._id,
 		showStyleVariantId: innerContext.showStyleCompound.showStyleVariantId,
 		bucketId,
 		importVersions,
+		uniquenessId: uniquenessId ?? unprotectString(id),
```

Comment:
I dont think this should be forced here,
uniquenessId is something the blueprints can set in their types, so we should respect that value (if they have set it). But I wouldnt be opposed to filling in a default value to match the externalId if it hasnt been set